### PR TITLE
Silence -Wshadow and -Wmaybe-uninitialized warnings.

### PR DESCRIPTION
**Pull request by rohit-n**

### CODE REVIEW

File: Source/Core/Core/Boot/Boot.cpp [hunk 1]
```diff
@@ -61,6 +61,7 @@ void CBoot::Load_FST(bool _bIsWii)
 		shift = 2;
 
 	u32 fst_offset, fst_size, max_fst_size;
+	fst_offset = fst_size = max_fst_size = 0;
```
Comment (JosJuice) [inline Source/Core/Core/Boot/Boot.cpp]: Give each variable declaration a line of its own instead:

    u32 fst_offset = 0;
    u32 fst_size = 0;
    u32 max_fst_size = 0;

File: Source/Core/Core/Boot/Boot.cpp [hunk 2]
```diff
@@ -61,6 +61,7 @@ void CBoot::Load_FST(bool _bIsWii)
 		shift = 2;
 
 	u32 fst_offset, fst_size, max_fst_size;
```
Comment (lioncash) [inline Source/Core/Core/Boot/Boot.cpp]: please initialize these individually.

File: Source/Core/DiscIO/VolumeCreator.cpp [hunk 3]
```diff
@@ -128,7 +128,7 @@ static std::unique_ptr<IVolume> CreateVolumeFromCryptedWiiImage(std::unique_ptr<
 
 	struct SPartition
 	{
-		SPartition(u64 offset, u32 type) : offset(offset), type(type) {}
+		SPartition(u64 _offset, u32 _type) : offset(_offset), type(_type) {}
```
Comment (lioncash) [inline Source/Core/DiscIO/VolumeCreator.cpp]: prefixed underscores are reserved by the standard.

Comment (Alcaro) [inline Source/Core/DiscIO/VolumeCreator.cpp]: http://stackoverflow.com/questions/228783/what-are-the-rules-about-using-an-underscore-in-a-c-identifier says "All identifiers that begin with an underscore are always reserved for use as identifiers with file scope in both the ordinary and tag name spaces.", which seems to imply we can shadow them. (We can't have macros with names like that, though.)

Personally I'd keep it like that, to match Python, but your choice.

**Pull request merged**